7a_temp_coop_munge/src/data_parsers/parse_mo_usace.R

```diff
@@ -0,0 +1,427 @@
+
+    # Site IDs are not not backfilled in most of the data sets
+    backfill_site_ids(.) %>%
+
+    # Removes all partially populated row created by the
+    # previous two munge steps
+    remove_incomplete_rows(.) %>%
+    convert_date(.) %>%
+    dplyr::bind_rows(.)
+
+}
+
+parse_mo_usace_2018 <- function(filepath,
+                                keep_cols = c(1, 2, 4, 5), remove_sheets) {
+
+  # list all sheets in the df and read each one
+  dat <- read_usace_sheets(filepath) %>%
+    # column name case is inconsistent - picking a convention and sticking with it
+    sheet_cols_to_lower(.) %>%
+
+    # a subset of years have extra sheets that are not needed - it was quicker to look them up and kick them out
+    remove_extra_sheets(., remove_sheets) %>%
+
+    # 2018 requires a hard-coded grab because I don't have a great way to
+    # automatically differentiate Depth and Temp. 2018 always requires these cols
+    lapply(., function(df){df[ , keep_cols]}) %>%
+
+    # Column names are inconsistent both between workbooks and within workbooks.
+    # A little column validation happens here
+    lapply(., function(df){
+      names(df) <- c('site', 'date', 'depth', 'temp')
+      return(df)
+    }) %>%
+
+    # All datasets have a mix of data and column headers which causes all
+    # data to import as `chr`
+    coerce_numeric_columns(.) %>%
+
+    # Site IDs are not not backfilled in most of the data sets
+    backfill_site_ids(.) %>%
+
+    # Removes all partially populated row created by the
+    # previous two munge steps
+    remove_incomplete_rows(.) %>%
+    convert_date(.) %>%
+    dplyr::bind_rows(.)
+
+}
+
+parse_mo_usace_2010 <- function(filepath, keep_cols,
+                                subset_sheets = c('PT', 'RA'),
+                                subset_skip = 1, remove_sheets) {
+
+  # list all sheets in the df and read each one
+  dat <- read_usace_sheets(filepath, subset_sheets, subset_skip) %>%
+    # column name case is inconsistent - picking a convention and sticking with it
+    sheet_cols_to_lower(.) %>%
+
+    # a subset of years have extra sheets that are not needed - it was quicker to look them up and kick them out
+    remove_extra_sheets(., remove_sheets) %>%
+
+    # Column names are inconsistent bot between workbooks and within workbooks.
+    # A little column validation happens here
+    validate_col_names(.) %>%
+
+    # Because of the data format, many extra columns come along
+    # for the ride. This step limits us to our columns of interest
+    select_cols_to_keep(., keep_cols) %>%
+
+    # All datasets have a mix of data and column headers which causes all
+    # data to import as `chr`
+    coerce_numeric_columns(.) %>%
+
+    # Site IDs are not not backfilled in most of the data sets
+    backfill_site_ids(.) %>%
+
+    # Removes all partially populated row created by the
+    # previous two munge steps
+    remove_incomplete_rows(.) %>%
+    convert_date(.) %>%
+    dplyr::bind_rows(.)
```

Comment:
Naming conventions for `sites` in this dataset are highly variable - both within years and between years _and_ when compared to the spatial data provided. Because the issues with names in the `site` column are so widespread I opted to clean up the `site` names on all datasets at the end of the parser ([link](https://github.com/USGS-R/lake-temperature-model-prep/blob/e82c8b9f1dd73a4715bbc0f17b43ea5ccc6e9861/7a_temp_coop_munge/src/data_parsers/parse_mo_usace.R#L74-L77)). That being said, I am going to translate the that snippet of code into tidyverse as requested a bit earlier in the review ([link](https://github.com/USGS-R/lake-temperature-model-prep/pull/311#discussion_r817126683)).

I **think** I have answered your question, but please let me know if I have not! 😅